client/runtime/src/main/java/io/quarkiverse/openapi/generator/providers/ConfigCredentialsProvider.java

```diff
@@ -63,4 +63,10 @@ public String getBearerToken(ClientRequestContext requestContext, String openApi
                 .orElse("");
     }
 
+    @Override
+    public void setOauth2BearerToken(ClientRequestContext requestContext, String accessToken, String openApiSpecId,
```

Comment:
In this case, you can create another interface that is a child of `CredentialsProvider` and add this method only to that interface.

Reply:
I just had an epiphany: if we are going to get the token previously set from the headers, then we can just use a `get` method.
After the `filter` is called, I call the `get` method from the credential provider and I store the token in a variable, as it is done in the others providers
Then at the end of the method, I set the header with this variable. In case token propagation is enabled, there is no way to override it (as for the other providers) but otherwise users can override what was set by the delegate.

And the default `get` method is simply retrieving the token set in the authorization header.

I did that using gh codespace and the format command is failing, hence the failing CI, if you like this approach I will fix it tomorrow (sorry for the live brainstorming)